### PR TITLE
[fix]: [PL-25105]: improve api description for framework squad

### PR DESCRIPTION
**Pull request by shreyas-harness**
## Describe your changes
Improved description for API's
## Checklist
- [x] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34554)

### CODE REVIEW

**Review by VikasMaddukuriHarness — APPROVED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34554)* status: 6 file(s) to review. 1 of 2 approvals obtained.

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/OrganizationResource.java [hunk 1]
```diff
@@ -158,8 +160,8 @@ public class OrganizationResource {
 
   @GET
   @ApiOperation(value = "Get Organization list", nickname = "getOrganizationList")
-  @Operation(operationId = "getOrganizationList",
-      summary = "Get the list of Organizations satisfying the criteria (if any) in the request",
+  @Operation(operationId = "getOrganizationList", summary = "List all Organizations using a filter",
+      description = "Lists all Organizations matching search criteria.",
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/OrganizationResource.java:164]: List all Organizations matching the given search criteria.

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/OrganizationResource.java [hunk 2]
```diff
@@ -158,8 +160,8 @@ public class OrganizationResource {
 
   @GET
   @ApiOperation(value = "Get Organization list", nickname = "getOrganizationList")
-  @Operation(operationId = "getOrganizationList",
-      summary = "Get the list of Organizations satisfying the criteria (if any) in the request",
+  @Operation(operationId = "getOrganizationList", summary = "List all Organizations using a filter",
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/OrganizationResource.java:163]: List Organizations by filter

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/ProjectResource.java [hunk 3]
```diff
@@ -259,7 +261,8 @@ public class ProjectResource {
   @PUT
   @Path("{identifier}")
   @ApiOperation(value = "Update a Project by ID", nickname = "putProject")
-  @Operation(operationId = "putProject", summary = "Update Project by ID",
+  @Operation(operationId = "putProject", summary = "Update a Project",
+      description = "Updates a Project's settings for the given ID.",
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/ProjectResource.java:265]: Updates Project details for the given ID.

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/ProjectResource.java [hunk 4]
```diff
@@ -289,7 +292,8 @@ public class ProjectResource {
   @DELETE
   @Path("{identifier}")
   @ApiOperation(value = "Delete a Project by identifier", nickname = "deleteProject")
-  @Operation(operationId = "deleteProject", summary = "Deletes the Project corresponding to the specified Project ID.",
+  @Operation(operationId = "deleteProject", summary = "Delete a Project",
+      description = "Deletes a Project for the specified Project ID.",
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/ProjectResource.java:296]: Deletes a Project corresponding to the given ID.

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/TokenResource.java [hunk 5]
```diff
@@ -204,8 +208,8 @@ public ResponseDTO<TokenDTO> getToken(@QueryParam("tokenId") String tokenId) {
   @GET
   @Path("aggregate")
   @ApiOperation(value = "List tokens", nickname = "listAggregatedTokens")
-  @Operation(operationId = "listAggregatedTokens",
-      summary = "Fetches the list of Aggregated Tokens corresponding to the request's filter criteria.",
+  @Operation(operationId = "listAggregatedTokens", summary = "List all Tokens",
+      description = "Lists all Tokens matching search criteria.",
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/TokenResource.java:212]: Lists all the Tokens matching the given search criteria.

File: 120-ng-manager/src/main/java/io/harness/ng/serviceaccounts/resource/ServiceAccountResource.java [hunk 6]
```diff
@@ -124,8 +124,8 @@ public class ServiceAccountResource {
   @PUT
   @Path("{identifier}")
   @ApiOperation(value = "Update service account", nickname = "updateServiceAccount")
-  @Operation(operationId = "updateServiceAccount", summary = "Update Service Account",
-      description = "Updates the Service Account.",
+  @Operation(operationId = "updateServiceAccount", summary = "Update a Service Account",
+      description = "Update the details of the Service Account for the given Service Account ID.",
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/serviceaccounts/resource/ServiceAccountResource.java:128]: Updates details of the Service Account for the given Service Account ID.

File: 120-ng-manager/src/main/java/io/harness/ng/serviceaccounts/resource/ServiceAccountResource.java [hunk 7]
```diff
@@ -151,8 +151,8 @@ public class ServiceAccountResource {
   @DELETE
   @Path("{identifier}")
   @ApiOperation(value = "Delete service account", nickname = "deleteServiceAccount")
-  @Operation(operationId = "deleteServiceAccount", summary = "Delete Service Account",
-      description = "Deletes Service Account by ID",
+  @Operation(operationId = "deleteServiceAccount", summary = "Delete a Service Account",
+      description = "Deletes a Service Account for the given Service Account ID.",
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/serviceaccounts/resource/ServiceAccountResource.java:155]: Deletes a Service Account corresponding to the given Service Account ID.

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java [hunk 8]
```diff
@@ -169,8 +170,9 @@ public ConnectorResource(@Named("connectorDecoratorService") ConnectorService co
   @GET
   @Path("validateUniqueIdentifier")
   @ApiOperation(value = "Validate Identifier is unique", nickname = "validateTheIdentifierIsUnique")
-  @Operation(operationId = "validateTheIdentifierIsUnique",
-      summary = "Validate the Connector by Account Identifier and Connector Identifier",
+  @Operation(operationId = "validateTheIdentifierIsUnique", summary = "Test a Harness Connector",
+      description =
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:174]: Looks like there is extra space at the beginning.

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java [hunk 9]
```diff
@@ -194,8 +196,8 @@ public ConnectorResource(@Named("connectorDecoratorService") ConnectorService co
 
   @GET
   @ApiOperation(value = "Gets Connector list", nickname = "getConnectorList")
-  @Operation(operationId = "getConnectorList",
-      summary = "Fetches the list of Connectors corresponding to the request's filter criteria.",
+  @Operation(operationId = "getConnectorList", summary = "List all Connectors using filters",
+      description = "Lists all Connectors matching specified filters.",
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:200]: Lists all the Connectors matching the specified filters.

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java [hunk 10]
```diff
@@ -443,7 +451,8 @@ public ConnectorResource(@Named("connectorDecoratorService") ConnectorService co
   @GET
   @Path("catalogue")
   @ApiOperation(value = "Get Connector Catalogue", nickname = "getConnectorCatalogue")
-  @Operation(operationId = "getConnectorCatalogue", summary = "Gets the Connector catalogue by Account Identifier",
+  @Operation(operationId = "getConnectorCatalogue", summary = "Lists all Connectors for an account",
+      description = "Lists all Connectors for an Account using an account ID.",
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:455]: Lists all the Connectors for the given Account ID.

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java [hunk 11]
```diff
@@ -481,8 +490,8 @@ public ConnectorResource(@Named("connectorDecoratorService") ConnectorService co
   @POST
   @Path("/listbyfqn")
   @ApiOperation(value = "Gets Connector list", nickname = "listConnectorByFQN")
-  @Operation(operationId = "listConnectorByFQN",
-      summary = "Get the list of connectors by FQN satisfying the criteria (if any) in the request",
+  @Operation(operationId = "listConnectorByFQN", summary = "List all Connectors for an account",
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:493]: The summary is the same as above?

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java [hunk 12]
```diff
@@ -540,7 +549,8 @@ public ResponseDTO<ConnectorValidationParameterResponse> getConnectorValidationP
   @GET
   @Path("/fieldValues")
   @ApiOperation(value = "Get All Allowed field values for Connector Type", nickname = "getAllAllowedFieldValues")
-  @Operation(operationId = "getAllAllowedFieldValues", summary = "Get the allowed field values by Connector Type",
+  @Operation(operationId = "getAllAllowedFieldValues", summary = "List all settings for a Connector type",
+      description = "Returns the field values that are allowed for each Connector type.",
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:553]: Returns the list of Connector settings for the given Connector type.

File: 955-filters-sdk/src/main/java/io/harness/filter/resource/FilterResource.java [hunk 13]
```diff
@@ -103,8 +104,8 @@ public class FilterResource {
 
   @GET
   @ApiOperation(value = "Get Filter", nickname = "getFilterList")
-  @Operation(operationId = "getConnectorListV2",
-      summary = "Get the list of Filters satisfying the criteria (if any) in the request",
+  @Operation(operationId = "getConnectorListV2", summary = "Lists filters",
+      description = "Lists filters using specified criteria such as type.",
```
Comment (rashminandasahoo) [inline 955-filters-sdk/src/main/java/io/harness/filter/resource/FilterResource.java:108]: Lists filters for the given criteria.

Is filter a Harness entity? If yes then the casing should be "Filter".

**Review by rashminandasahoo — CHANGES_REQUESTED** [review]

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java (hunk 11 above)
Comment (shreyas-harness) [reply 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:493]: Thats what is mentioned in the sheet you shared. - https://docs.google.com/spreadsheets/d/13jCy23e4gBr5qUMTydZ4BhH2yQ3mtqqX9XzxGrkWMhY/edit#gid=0

Comment (rashminandasahoo) [reply 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:493]: I meant to ask if we have 2 APIs with the same name (summary)?

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/OrganizationResource.java (hunk 1 above)
Comment (shreyas-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/core/remote/OrganizationResource.java]: Done

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/OrganizationResource.java (hunk 2 above)
Comment (shreyas-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/core/remote/OrganizationResource.java]: Done

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/ProjectResource.java (hunk 3 above)
Comment (shreyas-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/core/remote/ProjectResource.java]: Done

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/ProjectResource.java (hunk 4 above)
Comment (shreyas-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/core/remote/ProjectResource.java]: Done

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/TokenResource.java (hunk 5 above)
Comment (shreyas-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/core/remote/TokenResource.java]: Done

File: 120-ng-manager/src/main/java/io/harness/ng/serviceaccounts/resource/ServiceAccountResource.java (hunk 6 above)
Comment (shreyas-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/serviceaccounts/resource/ServiceAccountResource.java]: Done

File: 120-ng-manager/src/main/java/io/harness/ng/serviceaccounts/resource/ServiceAccountResource.java (hunk 7 above)
Comment (shreyas-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/serviceaccounts/resource/ServiceAccountResource.java]: Done

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java (hunk 8 above)
Comment (shreyas-harness) [reply 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:174]: This isn't the case.

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java (hunk 9 above)
Comment (shreyas-harness) [reply 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java]: Done

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java (hunk 10 above)
Comment (shreyas-harness) [reply 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java]: Done

File: 955-filters-sdk/src/main/java/io/harness/filter/resource/FilterResource.java (hunk 13 above)
Comment (shreyas-harness) [reply 955-filters-sdk/src/main/java/io/harness/filter/resource/FilterResource.java]: Done.

**Review by rashminandasahoo — APPROVED** [review]
:lgtm:

> Reviewed 5 of 6 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34554#-:-N5J4W1U6EEmNJjlskJh:bnfp4nl)* status: 6 file(s) to review. 1 of 2 approvals obtained.

**Review by VikasMaddukuriHarness — APPROVED** [review]
> Reviewed 5 of 6 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34554)* status: :shipit: complete! ✓

**Pull request merged**